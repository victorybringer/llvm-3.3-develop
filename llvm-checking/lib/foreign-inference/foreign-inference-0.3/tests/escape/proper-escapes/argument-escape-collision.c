extern int* global;

typedef void (*fptr)(int*);

extern fptr indirectCallee;

void g(int *p, int *q)
{
  global = p;
  indirectCallee(q);
}

void f(int *i1, int *i2)
{
  g(i1, i2);
}
