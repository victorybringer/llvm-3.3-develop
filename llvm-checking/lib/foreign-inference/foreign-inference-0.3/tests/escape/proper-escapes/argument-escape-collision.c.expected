fromList [ ("@g", fromList [ (DirectEscape, "%p"), (IndirectEscape, "%q") ] )
         , ("@f", fromList [ (DirectEscape, "%i1"), (IndirectEscape, "%i2") ] )
         ]